### PR TITLE
Fix blank user name

### LINKED ISSUE #19617: Blank user-name
**Describe the bug**
Apparently, it is possible to have a user-name (person's name) that is blank.

**To Reproduce**
1 - Go to this link: https://dev.to/pas_27513 and check this user-name, or also this link https://dev.to/theaccordance/the-reddit-blackout-is-a-lesson-in-risk-k5e and checkout the comments

**Expected behavior**
Not completely sure if this is allowed, but I think it would be expected that every user had a name displayed in their profiles and where they comment, so not completely sure if it's a bug or if it's allowed.

**Screenshots**
image
image
image

**Desktop:**

- OS, version: Linux Mint 21.1
- Browser, version: Firefox 113.0.1 (64-bit) and Google Chrome 113.0.5672.126 (Official Build) (64-bit)

**Additional context**
I use “user-name” because I refer not to the username, I refer to the person's name.

Also, I'm sending this bug as requested by the reply from the yo@dev.to where I first submitted this issue.

**Requested change**

- When a user signs up or edits their profile, we should ensure that their name field does not contain only whitespace.
- If a user tries to save their profile, and their name field contains only whitespace - we should return the existing error message stating "Name is too short"
- This could be accomplished by validating using a `trim()` method or something similar.

### PR DESCRIPTION
**Pull request by jerinkc**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
- Presence validation for name and corresponding fixes in the UI

## Related Tickets & Documents

- Related Issue #19617 (does not close due to remaining invalid data)

## QA Instructions, Screenshots, Recordings
1. Update the "Name" field of profile settings (settings -> profile)
2. Submit the form
Result: Displays error message "Name can't be blank" message.

Recording: https://www.loom.com/share/90c1e18bdb7a4621a03888d401d1317e?sid=67e95f38-8d56-4540-ac49-54d73528d02d

### UI accessibility checklist
_If your PR includes UI changes, please utilize this checklist:_
- [ ] Semantic HTML implemented?
- [ ] Keyboard operability supported?
- [ ] Checked with [axe DevTools](https://www.deque.com/axe/) and addressed `Critical` and `Serious` issues?
- [ ] Color contrast tested?

## Added/updated tests?

- [x] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

**Comment by rt4914** [comment]
Thanks for the PR @jerinkc . @maestromac and @lightalloy will review this PR soon. Thanks.

**Comment by PhilipHow** [comment]
@jerinkc, thanks so much for this PR - it's really appreciated.

> Redirecting users with blank name to profile settings page.

This is not something I was expecting as part of this change. Firstly my apologies for not being clearer in the issue to detail what we were expecting here - but in reality, it was only the validation part of the PR.

Could you remove this from your solution?

**Review by lightalloy — COMMENTED** [review]
Hey, @jerinkc , thanks for the pr!
As we have decided that forcing users to update their `name` via redirect shouldn't be a part of this change, the changes related to displaying error messages are also not relevant.

The validation you have added is necessary, but it will leave some users in invalid state, I will discuss potential solutions to this problem with the team.

**Comment by jerinkc** [comment]
Hey @lightalloy, Thanks for your feedback. Based on my understanding, you would like me to remove the error messages that appear on other tabs, but not the ones that show up on the form. If that is correct, I have already implemented this change and updated the code accordingly.

**Comment by jerinkc** [comment]
Hey @lightalloy,

I apologize for the inadvertent changes related to the translation file. I appreciate your feedback and have reverted them in the latest commits.

Regarding the changes in `app/views/users/_errors.html.erb`, I have decided to keep them as they fix the current issue of displaying redundant and incorrect error messages, such as '1 error 1 error prohibited your profile from being saved::' or '2 errors 2 errors prohibited your profile from being saved::'.

I have also implemented the data update scripts to use username as the name, as you suggested. Thank you for this brilliant solution and documentation.

File: lib/data_update_scripts/20230718102644_backfill_user_blank_name.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+module DataUpdateScripts
+  class BackfillUserBlankName
+    def run
+      User.where(" TRIM(name)='' ").each do |user|
+        titleized_username = user.username.titleize
```
Comment (lightalloy) [inline lib/data_update_scripts/20230718102644_backfill_user_blank_name.rb:5]: I think we can go without titleizing, we allow names starting with a lowercase letter.

**Review by lightalloy — COMMENTED** [review]
Thanks for implementing the changes :sparkles: 
Looks good, I left one small comment about titleizing

File: app/views/users/_errors.html.erb [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <div class="crayons-card crayons-card--secondary crayons-notice crayons-notice--danger" data-testid="account-errors-panel" role="alert">
   <div class="crayons-card__header">
     <h3 class="crayons-card__headline">
-      <%= pluralize(errors.size, "error") %> <%= title_suffix %>:
+      <%= title_suffix %>
```
Comment (lightalloy) [inline app/views/users/_errors.html.erb:4]: Thanks for fixing the bug along the way :raised_hands:

File: lib/data_update_scripts/20230718102644_backfill_user_blank_name.rb (hunk 1 above)
Comment (jerinkc) [reply lib/data_update_scripts/20230718102644_backfill_user_blank_name.rb:5]: No titleize, got it

**Review by maestromac — APPROVED** [review]
Thank you for this PR. LGTM!

**Review by lightalloy — APPROVED** [review]

**Pull request merged**